### PR TITLE
Berry Smoothness and other details for future uses

### PR DESCRIPTION
**Pull request by osherwott**
### New Berry Elements
_Values for all three elements were taken from Bulbapedia, with the exception of Pinkan, Maranga, and Petaya, who do not have canon characteristics_
- **_Berry smoothness_** based off of gen 4 values, with only Lum boosted to reflect its rarer Pokeclicker status. Gen 4 smoothness values were chosen over gen 3 as they increase more gradually over the longer list of berries. This also allows for the eReader berries, who only have a high gen 3 value, to have a high status if added in the future. Pinkan, Maranga, and Petaya have no canon smoothness and their values are based on those of the surrounding berries. This value will be used for pokeblock creation which is part of contests.
- **_Berry sizes_**, per canon. Pinkan's is based on Oran's, while Maranga and Petaya's are taken from their real world inspiration's counterpart. Sizes do not have a current use but can be planned to later.
- **_Berry firmness_**. Same explanation as sizes. There is currently an idea in the works for using firmness, but no attempt will be made to implement anything until a later date.
### New Files
- **_SizeUnits.ts_** enum: inches or centimeters
- **_BerryFirmness.ts_** enum: ranging from Very Soft to Super Hard
### Existing code expanded upon
- **_BerryDex Rows for Color, Size, Firmness, and Current Amount_**
cheri image
- **_Setting to switch displayed size_** in BerryDex between inches or centimeters
- **_Gold and Silver berry colors and wanderers_** as placeholders in preparation for the Let's Go berries, or any other berry that could be classified as such. These can be changed if needed.
image
- **_Pokeblock colors in GameConstants_** to complete the existing list
_Note: This is assuming the pre-existing data for pokeblocks was meant to be utilized and expanded upon._
- Added original Flaxen named pokeblock color to distinguish between different tiered Gold ones and make pokeblock implementation easier. (In the games pokeblocks with a flavor profile of over 50 are named "Gold" regardless of the flavor combination. This change will make it easier to identify the amount of flavors present in a pokeblock, or allow for a different implementation by still having relegated colors for each combination of flavors. The word "flaxen" was chosen to match the theme of color lightening with every additional flavor and can be changed to something else if needed.)

### CODE REVIEW

File: src/scripts/GameConstants.d.ts [hunk 1]
```diff
@@ -70,6 +70,7 @@ namespace GameConstants {
     declare const FREEZE_MULCH_MULTIPLIER: number;
     declare const WANDER_RATE: number;
     declare const BerryColor: string[];
+    declare const BerryFirmness: string[];
```
Comment (Aegyo) [inline src/scripts/GameConstants.d.ts:73]: BerryFirmness doesn't exist in game constants, the BerryColor array I think you are imitating is actually the array of hex color codes, not the enum. No need for a manual declaration, your /enums/BerryFirmness.ts will generate one.

**Review by Aegyo — COMMENTED** [review]
Do you have a link to where you took the data for each berry?

**Comment by osherwott** [comment]
All official data was gathered from Bulbapedia. Size and Firmness have no use in-game which warrants a separate page like Smoothness, so the numbers were taken from each berry's individual page. A list of gen 3 berries is available from another site to easily confirm some of them is linked below, if needed.

Smoothness list (only Lum is different due to its status in Pokeclicker)
https://bulbapedia.bulbagarden.net/wiki/Smoothness
Size and Firmness list (only for berries up to gen 3 games)
https://strategywiki.org/wiki/Pok%C3%A9mon_Ruby_and_Sapphire/Berries

Roseli, Maranga, and Kee's sizes are taken from a random, average-ish number within the range of the fruit's size in their respective real world counterparts' wikipedia page (linked in Bulbapedia)
Their firmness is based off their real world counterparts' _exterior_ as researched (see: Custap being classified as Super Hard in the games despite its soft interior)
Smoothness is taken from the surrounding berries
Note: Roseli's Bulbapedia page has unsourced data for these. Berry tags and smoothness disappeared from the games after gen 4, so their section is viewed as non-credible.

Pinkan's size is taken from Oran's, as they appear to be similarly sized in the anime. Smoothness is matched to the surrounding berries. Firmness is the only value with no basis, but it was chosen to be slightly softer than Oran's.

**Comment by Aegyo** [comment]
I don't understand the Lum smoothness exception, could you write something I can point people at if they ever ask why it is different?

File: src/scripts/farming/Farming.ts [hunk 2]
```diff
@@ -205,7 +235,10 @@ class Farming implements Feature {
             20,
             2,
             [0, 10, 10, 0, 0],
+            20,
             BerryColor.Purple,
+            10.8,
```
Comment (Aegyo) [inline src/scripts/farming/Farming.ts:240]: Bluk is 4.7" on your link, but displays as 4.3 with this size

File: src/scripts/farming/Farming.ts [hunk 3]
```diff
@@ -277,7 +325,10 @@ class Farming implements Feature {
             50,
             3,
             [0, 0, 15, 0, 0],
+            25,
             BerryColor.Pink,
+            12.6,
```
Comment (Aegyo) [inline src/scripts/farming/Farming.ts:330]: Mago is 4.7" in your link, displaying in game as 5.0

File: src/scripts/farming/Farming.ts [hunk 4]
```diff
@@ -971,7 +1163,10 @@ class Farming implements Feature {
             15000,
             20,
             [10, 30, 0, 0, 30],
+            40,
             BerryColor.Blue,
+            7.5,
+            BerryFirmness.Very_Hard,
```
Comment (Aegyo) [inline src/scripts/farming/Farming.ts:1169]: Apicot is listed as only Hard in your link

**Review by Aegyo — COMMENTED** [review]
Thanks for the links, I found a couple of things that don't match but lmk if they are different for a reason :+1:

**Comment by osherwott** [comment]
Lum has an extra 5 smoothness in Pokeclicker because it's somewhat rarer in Pokeclicker. I don't think it will make a big difference since it's such a tiny boost, but it feels like the right choice that better reflects its status in this game (even if a bit). I also think it's easier to accept it becoming "better", as a higher smoothness is good for pokeblocks.

### And to futureproof:
_Pokeblocks have still not begun development, so if Lum ends up overshadowing other berries, there is a possibility its smoothness will be reverted back to canon, as the goal of this PR is to give relevance to berries that aren't currently useful._

File: src/scripts/farming/Farming.ts (hunk 2 above)
Comment (osherwott) [reply src/scripts/farming/Farming.ts:240]: Seems as though that link wasn't entirely reliable. Bulbapedia, Serebii, and its in-game Berry Tag in both gen 3 and 4 list it as 4.3.

File: src/scripts/farming/Farming.ts (hunk 3 above)
Comment (osherwott) [reply src/scripts/farming/Farming.ts:330]: Same resolution as Bluk, though I can only find a screenshot of its gen 3 tag (which is good enough, I believe).

File: src/scripts/farming/Farming.ts (hunk 4 above)
Comment (osherwott) [reply src/scripts/farming/Farming.ts:1169]: Serebii and its gen 3 berry tag says Hard, so this will be changed.

**Comment by DraKay** [comment]
Wich games exactly are you taking as some stats change between XY and ORAS afaik at least some colors change (qualot is Canon pink in oras and magost is green, but that would need a very deep talk before even think about make changes)

**Comment by osherwott** [comment]
All these values are from gen 4 (Diamond, Pearl, and Platinum). Nothing is being changed, only new data is added. I am currently double checking everything with Serebii and then again with these screenshots from gen 3 (Ruby, Sapphire, and Emerald) linked below:
https://pkmn.net/?action=content&page=viewpage&id=4244

**Comment by osherwott** [comment]
The added values have been double checked with
https://www.serebii.net/itemdex/list/berry.shtml

Ready to merge!

**Review by Aegyo — APPROVED** [review]

**Pull request merged**